File: Implementation/10774.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
	int j, a;
	cin >> j >> a;
	vector<int> jsize; // S:0, M:1, L:2
	jsize.push_back(0);
	for(int i = 0; i < j; i++){
		char sz;
		cin >> sz;
		if(sz == 'S') jsize.push_back(0);
		else if(sz == 'M') jsize.push_back(1);
		else jsize.push_back(2);
	}
	vector<bool> used(jsize.size());
	int answer = 0;
	for(int i = 0; i < a; i++){
		char sz; int num;
		cin >> sz >> num;
		int nsz;
		if(sz == 'S') nsz = 0;
		else if(sz == 'M') nsz = 1;
		else nsz = 2;
		if(jsize[num] >= nsz && !used[num]) {
			used[num] = true;
			answer++;
		}
	}
	cout << answer;
	return 0;
}

File: Implementation/10774.js
const fs = require('fs');
const [N, M, ...input] = fs.readFileSync('/dev/stdin', 'utf8').split('\n');
const n = Number(N);
const m = Number(M);

let size = [0, ];
for(let i = 0; i < n; i++){
	if(input[i] == 'S') size.push(0);
	else if(input[i] == 'M') size.push(1);
	else size.push(2);
}

let used = Array(size.length).fill(false);
let answer = 0;
for(let i = n; i < n + m; i++){
	let [tp, nb] = input[i].split(' ');
	let num = Number(nb);
	let sz;
	if(tp == 'S') sz = 0;
	else if(tp == 'M') sz = 1;
	else sz = 2;
	
	if(size[num] >= sz && !used[num]){
		used[num] = true;
		answer++;
	}
}
console.log(answer);
